mongo/database.go

```diff
@@ -552,6 +552,25 @@ func (db *Database) CreateCollection(ctx context.Context, name string, opts ...*
 		}
 		op.Validator(validator)
 	}
+	if cco.TimeSeriesOptions != nil {
+		idx, doc := bsoncore.AppendDocumentStart(nil)
+		doc = bsoncore.AppendStringElement(doc, "timeField", cco.TimeSeriesOptions.TimeField)
+
+		if cco.TimeSeriesOptions.MetaField != nil {
+			doc = bsoncore.AppendStringElement(doc, "metaField", *cco.TimeSeriesOptions.MetaField)
+		}
+
+		if cco.TimeSeriesOptions.ExpireAfterSeconds != nil {
+			doc = bsoncore.AppendInt32Element(doc, "expireAfterSeconds", *cco.TimeSeriesOptions.ExpireAfterSeconds)
```

Comment:
Is there any guidance in the server scope/design as to whether this should be an int32 or int64? I don't think there is any practical difference.

It looks like the server will store this as an int64 – I tried both ways:

```
> db.runCommand({create: "coll", timeseries: {timeField: "foo", expireAfterSeconds: 123}})
```

And:
```
db.runCommand({create: "coll", timeseries: {timeField: "foo", expireAfterSeconds: NumberLong(123)}})
```

Based on the output of `db.runCommand({listCollections: 1, filter: {name: "system.buckets.coll"}})`, both create the bucket collection with a `NumberLong` `expireAfterSeconds`.

If int64 is the way to go, I think it'd be useful to be prescriptive for other drivers (e.g. if where PHP assumes an int64, but libmongoc assumes an int32, it could require casting). The spec test could be tweaked to use `$numberLong`.

Reply:
From what you've said and my own testing, I think it will be better to use an `int64`. Updates to the spec are being made in [DRIVERS-1774](https://jira.mongodb.org/browse/DRIVERS-1774), and I've updated the code here and elsewhere to expect an `int64`.

Reply:
I was linked here from https://github.com/mongodb/specifications/pull/1003#discussion_r640883690

Assuming the server accepts 32-bit or 64-bit integers, I think it makes sense for Go to either:

 * Always encode a 64-bit integer (simple approach)
 * Conditionally encode a 32-bit or 64-bit integer based on the value

I disagreed with using `$numberLong` in the spec test's `operation.arguments` field as that really has no bearing on how drivers implement the API. It's only going to force the Extended JSON used to read the test file as a 64-bit integer. In PHP's case, the 64-bit integer type is going to end up holding a value in the 32-bit range and we're going to encode that as a 32-bit integer for BSON regardless.